### PR TITLE
Delete webinar permanantly

### PR DESCRIPTION
**Pull request by ChathurikaThathsarani**

### CODE REVIEW

File: src/api/controllers/Webinar.controller.ts [hunk 1]
```diff
@@ -193,3 +193,14 @@ export const recoverRemovedWebinar = async (request: Request, response: Response
  *@todo implement the @function deleteWebinarPermanently
  * @param webinarId @type string
  */
+export const deleteWebinarPermanently = async (request: Request, response: Response, next: NextFunction) => {
+	await WebinarService.deleteWebinarPermanently(request.body.webinarId)
```
Comment (senuravihanjayadeva) [inline src/api/controllers/Webinar.controller.ts:197]: check first request.body.webinarId is existing or not

**Review by senuravihanjayadeva — CHANGES_REQUESTED** [review]

File: src/api/services/index.ts [hunk 2]
```diff
@@ -37,6 +37,7 @@ import {
 	getAllWebinarsForAdmin,
 	getDeletedWebinarsForAdmin,
 	recoverDeletedWebinar,
+	deleteWebinarPermanently,
 	//@todo import deleteWebinarPermanently method
```
Comment (miyurugunarathna) [inline src/api/services/index.ts:41]: Remove this todo.

File: src/api/services/index.ts [hunk 3]
```diff
@@ -130,6 +131,7 @@ export default {
 	getAllWebinarsForAdmin,
 	getDeletedWebinarsForAdmin,
 	recoverDeletedWebinar,
+	deleteWebinarPermanently,
 	//@todo export deleteWebinarPermanently method
```
Comment (miyurugunarathna) [inline src/api/services/index.ts:135]: Remove this todo.

File: src/api/services/Webinar.service.ts [hunk 4]
```diff
@@ -243,3 +243,17 @@ export const recoverDeletedWebinar = async (webinarId: string) => {
  *@todo implement the @function deleteWebinarPermanently
  * @param webinarId @type string
  */
+
+export const deleteWebinarPermanently = async (webinarId: string) => {
```
Comment (miyurugunarathna) [inline src/api/services/Webinar.service.ts:247]: Use async await properly to this function.

**Review by miyurugunarathna — CHANGES_REQUESTED** [review]

File: src/api/services/Webinar.service.ts [hunk 5]
```diff
@@ -243,3 +243,17 @@ export const recoverDeletedWebinar = async (webinarId: string) => {
...
+	if (webinarId) {
+		return WebinarModel.findByIdAndDelete(webinarId)
```
Comment (senuravihanjayadeva) [inline src/api/services/Webinar.service.ts:249]: use await here

File: src/interfaces/IWebinar.ts [hunk 6]
```diff
@@ -20,4 +20,4 @@ interface IWebinar extends Document {
 	deletedBy?: Schema.Types.ObjectId | null;
 }
 
-export type { IWebinar };
+export type { IWebinar, IUpdatedBy };
```
Comment (senuravihanjayadeva) [inline src/interfaces/IWebinar.ts:23]: No need to exported IUpdatedBy here.  Exporting IWebinar is enough

**Review by senuravihanjayadeva — CHANGES_REQUESTED** [review]

File: src/api/services/index.ts (hunk 3 above)
Comment (senuravihanjayadeva) [reply src/api/services/index.ts:134]: Remove this todo.

**Review by senuravihanjayadeva — CHANGES_REQUESTED** [review]

File: src/api/controllers/Webinar.controller.ts [hunk 7]
```diff
@@ -190,6 +190,21 @@ export const recoverRemovedWebinar = async (request: Request, response: Response
 
 /**
  delete an webinar in the system
- *@todo implement the @function deleteWebinarPermanently
  * @param webinarId @type string
  */
+export const deleteWebinarPermanently = async (request: Request, response: Response, next: NextFunction) => {
+	const webinarId = request.body.webinarId;
+	if (webinarId) {
+		await WebinarService.deleteWebinarPermanently(request.body.webinarId)
```
Comment (senuravihanjayadeva) [inline src/api/controllers/Webinar.controller.ts:198]: use webinarId instead request.body.webinarId

**Review by senuravihanjayadeva — CHANGES_REQUESTED** [review]

**Review by KaveenLaksitha — APPROVED** [review]

**Review by KaveenLaksitha — APPROVED** [review]

**Pull request merged**